### PR TITLE
Document alternate locale handling

### PR DESCRIPTION
**Pull request by dbu**
continuation of #691

| Q             | A
| ------------- | ---
| Doc fix?      | no
| New docs?     | yes
| Applies to    | SeoBundle
| Fixed tickets | #605 

This will be my first try to describe the alternate locales. I know the english is bad, but it can be a base for the documentation i hope.

Tasks:
* [x] configuration page changes
* [x] grammar review
* [x] chapter in intro

### CODE REVIEW

**Comment by dbu** [comment]
worked on the open issues from #691

@ElectricMaxxx do you want to review this one?

**Comment by ElectricMaxxx** [comment]
Except the errors on travis it looks good to me.

**Comment by WouterJ** [comment]
travis failure is not related to this PR. It seems like the spell checker builder has trouble parsing the config options related to fabien's Sphinx extension.

File: bundles/seo/alternate_locale.rst [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+=========================
+
+Alternate locales are a way of telling search engines how to find translations
+of the current page. The SeoBundle provides a way to manage alternate locales
+and output them together with the other SEO information.
+
+When the alternate locale handling is set up and found alternates, you will
+find links like the following in the ``<head>`` part of your HTML pages:
+
+.. code-block: html
```
Comment (WouterJ) [inline bundles/seo/alternate_locale.rst]: single colon should be a double one here

File: bundles/seo/alternate_locale.rst [hunk 2]
```diff
@@ -0,0 +1,106 @@
+Alternate Locale Handling
+=========================
+
+Alternate locales are a way of telling search engines how to find translations
+of the current page. The SeoBundle provides a way to manage alternate locales
+and output them together with the other SEO information.
```
Comment (WouterJ) [inline bundles/seo/alternate_locale.rst]: "output them" doesn't sound 100% correct to me. However, I can't come up with any better alternative ("displaying" doesn't have a correct meaning here)

File: bundles/seo/alternate_locale.rst [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+When the alternate locale handling is set up and found alternates, you will
+find links like the following in the ``<head>`` part of your HTML pages:
+
+.. code-block: html
+
+    <link rel="alternate" href="/fr/le-titre" hreflang="fr" />
+    <link rel="alternate" href="/de/der-titel" hreflang="de" />
```
Comment (WouterJ) [inline bundles/seo/alternate_locale.rst]: The docs use HTML5 code, so the `/` should be removed imo

File: bundles/seo/alternate_locale.rst [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+When using PHPCR-ODM, there is almost no work to do, as the bundle can use the
+Doctrine meta data to figure out which translations exists for a content. More
+information on translating content with the PHPCR-ODM is in the chapter
+:doc:`Doctrine PHPCR-ODM Multilanguage Support<../phpcr_odm/multilang>`.
```
Comment (WouterJ) [inline bundles/seo/alternate_locale.rst]: there should be a space between the label and `<..>`

File: bundles/seo/alternate_locale.rst [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+Setting Up Alternate Locale Support
+-----------------------------------
+
+Enable alternate locale support. If you are using PHPCR-ODM, also enable
+support for that:
```
Comment (WouterJ) [inline bundles/seo/alternate_locale.rst]: It's a bit vague what "that" refers to. Is it the `persistence: { phpcr: ~ }` setting or the `alternate_locale: ~` setting? If it's the latter, I would say "enable PHPCR persistance and the ``alternate_locale`` setting."

File: bundles/seo/alternate_locale.rst [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+                <persistence>
+                    <phpcr enabled="true"/>
+                </persistence>
+            </config>
+        </container>
+
+    .. code-block:: php
+
+        $container->loadFromExtension('cmf_seo', array(
+            'alternate_locale' => array ('enabled' => true),
```
Comment (WouterJ) [inline bundles/seo/alternate_locale.rst]: this can be `'alternate_locale' => true` as well.

File: bundles/seo/alternate_locale.rst [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+            persistence:
+                phpcr: ~
+
+    .. code-block:: xml
+
+        <!-- app/config/config.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services">
+            <config xmlns="http://cmf.symfony.com/schema/dic/seo">
+                <alternate-locale enabled="true" />
```
Comment (WouterJ) [inline bundles/seo/alternate_locale.rst]: `<alternate-locale />` seems a bit nicer IMO

File: bundles/seo/alternate_locale.rst [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    .. code-block:: php
+
+        $container->loadFromExtension('cmf_seo', array(
+            'alternate_locale' => array (
+                'provider_id' => 'alternate_locale.provider',
+            ),
+        ));
+
+.. versionadded:: 1.2
+    For activated :doc:`Sitemap<sitemap>` the alternate locales will be pushed into the sitemap too.
```
Comment (WouterJ) [inline bundles/seo/alternate_locale.rst]: ``rst
.. versionadded:: 1.2
    When enabled, alternate locales will also be added to the :doc:`Sitemap <sitemap>`.
```

File: bundles/seo/alternate_locale.rst [hunk 9]
```diff
@@ -0,0 +1,106 @@
+Alternate Locale Handling
+=========================
```
Comment (WouterJ) [inline bundles/seo/alternate_locale.rst]: missing versionadded for alternate locale

File: bundles/seo/configuration.rst [hunk 10]
```diff
@@ -353,3 +353,52 @@ routes, use:
...
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/config.yml
+        cmf_seo:
+            alternate_locale:
+                enabled: true
+                provider_id: acme.application.alternate_locale.provider
```
Comment (WouterJ) [inline bundles/seo/configuration.rst]: Let's use `app.alternate_locale_provider` to implement the official best practices.

File: bundles/seo/configuration.rst [hunk 11]
```diff
@@ -353,3 +353,52 @@ routes, use:
...
+**type**: ``boolean`` **default**: ``true``
+
+Whether or not the the :ref:`bundles-seo-alternate-locale` should be loaded
+
+``provider_id``
+"""""""""""""""
+
+**type**: ``string`` **default**: ``null``
+
+Define your a custom :doc:`AlternateLocaleProvider<../seo/alternate_locale>`
```
Comment (WouterJ) [inline bundles/seo/configuration.rst]: missing space before `<..>`

**Comment by WouterJ** [comment]
Nice article in general, thanks! Shouldn't this be merged into the 1.2 documentation (and then adding the notion about sitemaps in 1.3?)

File: bundles/seo/alternate_locale.rst (hunk 9 above)
Comment (ElectricMaxxx) [inline bundles/seo/alternate_locale.rst]: We should have a look into the changelog for that. I think it is available from 1.1 but i can he wrong.

File: bundles/seo/alternate_locale.rst (hunk 2 above)
Comment (dbu) [inline bundles/seo/alternate_locale.rst]: i'll write "render"

File: bundles/seo/alternate_locale.rst (hunk 9 above)
Comment (dbu) [inline bundles/seo/alternate_locale.rst]: added this, it was added in version 1.1 according to the changelog, and to the versionadded note in the configuration section

**Comment by dbu** [comment]
thanks a lot wouter for all that good feedback! i think i adressed all points.

strictly speaking yes, this could go into version 1.2. but not sure if we want to bother with that. i will squash the commits now, then if you want to pick it into 1.2 feel free to do so, otherwise lets just merge it into 1.3

**Comment by dbu** [comment]
cherry-picking this commit into the 1.2 branch leads to a lot of conflicts, so i decided to not bother with that.

**Pull request merged**